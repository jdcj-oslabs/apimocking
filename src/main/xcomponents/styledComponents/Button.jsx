import styled from 'styled-components';

let backgroundColor;

const Button = styled.button`
  background-color: ${(props) => {
    if (!props.enabled) return '#BCC1C2';
    if (props.variation === 'positive') return '#49893E';
    return '#1F4E7A';
  }};
  border: none;
  color: ${(props) => {
    if (!props.enabled) return '#555B5E';
    if (props.variation === 'positive') return '#DEF0DB';
    return '#D9EDF2';
  }};
  border-radius: 3px;
  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif;
  font-size: 1em;
  font-weight: 500;
  padding: 0.25em 1em;
  transition: all 0.15s ease-in-out;

  :hover {
    background-color: ${(props) => {
    if (!props.enabled) return '#BCC1C2';
    if (props.variation === 'positive') return '#35612E';
    return '#0D273F';
  }};
  }

  :active {
    color: #D4E6F7;
    outline: 0;
    box-shadow: none;
  }

  :focus {
    outline: 0;
    box-shadow: ${(props) => {
    if (!props.enabled) return '0 0 1px 2px #949C9E';
    if (props.variation === 'positive') return '0 0 1px 2px #DEF0DB';
    return '0 0 1px 2px #96C1E9';
  }}
  }
`;

export default Button;
