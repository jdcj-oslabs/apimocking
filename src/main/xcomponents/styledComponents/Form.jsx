import styled from 'styled-components';

const Form = styled.form`
  align-items: center;
  border: 1px solid #BCC1C2;
  border-radius: 3px;
  display: flex;
  flex-shrink: 1;
  justify-content: flex-start;
  margin-top: ${(props) => {
    if (!props.bordered) return '1em';
    return 'inherit';
  }}
  transition: all 0.15s ease-in-out
  &:focus-within {
    ${(props) => {
    if (props.bordered) {
      return `border: 1px solid #949C9E;
      box-shadow: 0 2px 6px -2px rgba(41, 47, 50, 0.25);`;
    }
    return 'none';
  }}
  }
  button {
    border-radius: 2px;
    margin-left: auto;
    margin-right: 0.5em;
    // min-width: 0;
  }
  input {
    min-width: 0;
  }
`;

export default Form;
