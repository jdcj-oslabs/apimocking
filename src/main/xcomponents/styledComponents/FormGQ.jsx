import styled from 'styled-components';

const Form = styled.form`
 color:white;
  border: 0px solid #BCC1C2;
  border-radius: 3px;
  display: grid;
  grid-gap: 15px;
  grid-template-columns: 600px;
  grid-template-rows: 50px 150px;
  padding: px;


  button {
    border-radius: 5px;
    color:white;
    border-radius: 2px;
    margin-left: auto;
    margin-right: 0.5em;
    min-width: 0;
  }
  textarea {
    border-radius: 5px;
    background-color: white;
    min-width: 0;
    border: none;
  border: 1px solid #D8D8D8;
  color: blue;
  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1em;
  line-height: 1em;
  }
  input {
    border-radius: 5px;
    background-color: white;
    min-width: 0;
    border: none;
  border: 1px solid #D8D8D8;

  color: blue;
  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1em;
  line-height: 1em;
  }
`;

export default Form;
