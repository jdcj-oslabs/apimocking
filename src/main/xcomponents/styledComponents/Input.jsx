import styled from 'styled-components';

const Input = styled.input`
  border: none;
  border-left: 1px solid #D8D8D8;
  border-bottom: 1px solid #D8D8D8;
  color: #292F32;
  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1em;
  line-height: 1em;
  padding: 0.825em;
  width: 100%
  &:focus {
    outline: none;
  }
  &::placeholder {
    color: #BCC1C2;
    font-size: 0.9em;
  }
`;

export default Input;
