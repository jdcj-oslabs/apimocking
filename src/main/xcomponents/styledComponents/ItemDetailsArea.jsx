import styled from 'styled-components';

const ItemDetailsArea = styled.div`
  display: flex;
  border: 1px solid #D8D8D8;
  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 1em;
  padding: 5px;
  flex-direction: column;
  .detail-pair {
    margin-bottom: 5px;
  }
  .detail-header {
    text-decoration: underline;
    font-weight: bold;
    color: black;
    text-align: center;
    width: 100%;
  }
  .detail-label {
    text-align: left;
    color: grey;
    font-size: 12px;
    font-style: italic;
  }
  .detail-value {
    text-align: left;
    color: grey;
    text-overflow: ellipsis;
    overflow: hidden;
    font-family: monospace;

    &:hover {
      z-index: 1000;
      background-color: white;
      overflow: visible;
      width: fit-content;
      position: sticky;
    }
  }
`;

export default ItemDetailsArea;
