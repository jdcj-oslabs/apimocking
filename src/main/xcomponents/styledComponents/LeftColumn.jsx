import styled from 'styled-components';

const LeftColumn = styled.div`
    display-flex;
    justify-content: center;
    border: 1px solid #D8D8D8;
    width: 30%;
    padding: 10px;
    height: 300px;
`;

export default LeftColumn;
