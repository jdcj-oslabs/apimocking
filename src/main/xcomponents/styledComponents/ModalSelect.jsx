import styled from "styled-components";

const ModalSelect = styled.select`
  align-items: left;
  justify-content: left;

  width: 10%;
  height: 35px;
  background: white;
  color: gray;
  padding-left: 5px;
  font-size: 14px;
  border: none;
  margin-left: 10px;

  option {
    color: black;
    background: white;
    display: flex;
    white-space: pre;
    min-height: 20px;
    padding: 0px 2px 1px;
  }
`;

export default ModalSelect;