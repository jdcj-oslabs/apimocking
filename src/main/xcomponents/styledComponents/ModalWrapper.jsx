import styled from 'styled-components';

const ModalWrapper = styled.div`
    border: 1px solid #D8D8D8;
    display: flex;
    flex-direction: column;
    width: inherited;  //
    border: 1px solid #D8D8D8;
    color: #292F32;
`;
export default ModalWrapper;