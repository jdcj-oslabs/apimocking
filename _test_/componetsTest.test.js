import toJson from 'enzyme-to-json'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Button from '../src/main/xcomponents/styledComponents/Button';
import Form from '../src/main/xcomponents/styledComponents/Form';
import FormGQ from '../src/main/xcomponents/styledComponents/FormGQ';
import Input from '../src/main/xcomponents/styledComponents/Input';
import ItemDetailsArea from '../src/main/xcomponents/styledComponents/ItemDetailsArea';
import LeftColumn from '../src/main/xcomponents/styledComponents/LeftColumn';
import ModalSelect from  '../src/main/xcomponents/styledComponents/ModalSelect';
import ModalWrapper from  '../src/main/xcomponents/styledComponents/ModalWrapper';
import ModelInput from '../src/main/xcomponents/styledComponents/ModelInput';
import RightColumn from '../src/main/xcomponents/styledComponents/RightColumn';
import Select from '../src/main/xcomponents/styledComponents/Select';
import TextArea from '../src/main/xcomponents/styledComponents/Textarea';
import WrapperDiv from '../src/main/xcomponents/styledComponents/WrapperDiv';


configure({adapter: new Adapter()})
test('it has a button', () => {
    const buttonWrapper = shallow(<Button />)
    expect(buttonWrapper).toMatchSnapshot()
})
test('for Form', () => {
    const formWrapper = shallow(<Form />)
    expect(formWrapper).toMatchSnapshot()
})
test('for FormGQ', () => {
    const formGQ = shallow(<FormGQ />)
    expect(formGQ).toMatchSnapshot()
})
test('for Input', () => {
    const input = shallow(<Input />)
    expect(input).toMatchSnapshot()
})

test('for ItemDetailsArea', () => {
    const itemDetailsArea = shallow(< ItemDetailsArea />)
    expect(itemDetailsArea).toMatchSnapshot()
})

test('for LeftColumn', () => {
    const leftColumn = shallow(<LeftColumn />)
    expect(leftColumn).toMatchSnapshot()
})

test('for ModalSelect', () => {
    const modalSelect = shallow(<ModalSelect />)
    expect(modalSelect).toMatchSnapshot()
})

test('for ModalWrapper', () => {
    const modalWrapper = shallow(<ModalWrapper />)
    expect(modalWrapper).toMatchSnapshot()
})

test('for ModelInput', () => {
    const modelInput = shallow(<ModelInput />)
    expect(modelInput).toMatchSnapshot()
})

test('for RightColumn', () => {
    const rightColumn = shallow(<RightColumn/>)
    expect(rightColumn).toMatchSnapshot()
})

test('for Select', () => {
    const select = shallow(<Select/>)
    expect(select).toMatchSnapshot()
})

test('for StyledModal', () => {
    const styledModal = shallow(<StyledModal />)
    expect(styledModal).toMatchSnapshot()
})

test('for TextArea', () => {
    const textArea = shallow(<TextArea />)
    expect(textArea).toMatchSnapshot()
})

test('for WrapperDiv', () => {
    const wrapperDiv = shallow(<WrapperDiv />)
    expect(wrapperDiv).toMatchSnapshot()
})
test('for WrapperLove', () => {
    const wrapperDiv = shallow(<WrapperDiv />)
    expect(wrapperDiv).toMatchSnapshot()
})





