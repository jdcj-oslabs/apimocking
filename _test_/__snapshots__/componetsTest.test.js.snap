// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for Form 1`] = `ShallowWrapper {}`;

exports[`for FormGQ 1`] = `ShallowWrapper {}`;

exports[`for FormGQ 2`] = `ShallowWrapper {}`;

exports[`for Input 1`] = `ShallowWrapper {}`;

exports[`for ItemDetailsArea 1`] = `ShallowWrapper {}`;

exports[`for LeftColumn 1`] = `ShallowWrapper {}`;

exports[`for ModalSelect 1`] = `ShallowWrapper {}`;

exports[`for ModalWrapper 1`] = `ShallowWrapper {}`;

exports[`for ModelInput 1`] = `ShallowWrapper {}`;

exports[`for RightColumn 1`] = `ShallowWrapper {}`;

exports[`for Select 1`] = `ShallowWrapper {}`;

exports[`for TextArea 1`] = `ShallowWrapper {}`;

exports[`for WrapperDiv 1`] = `ShallowWrapper {}`;

exports[`it has a button 1`] = `ShallowWrapper {}`;
